src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java

```diff
@@ -200,6 +200,14 @@ public boolean execute(
 
         // Use sample info type for each output band to estimate size
         List<CoverageDimensionInfo> coverageDimensionInfoList = coverageInfo.getDimensions();
+
+
+        if (coverageDimensionInfoList.stream().anyMatch(cdi -> cdi.getDimensionType() == null)) {
+            if (LOGGER.isLoggable(Level.FINE)) {
```

Comment:
The guard is not necessary when the message is a static string (no construction of new objects needed).

Reply:
Sorry Andrea, where is the construction of a new object in these lines?

Reply:
Indeed there is none ("no construction of new objects needed") so no need to guard  the statement